### PR TITLE
Obsession Solution Rework: There's Options

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

Obsessed is no longer "cured" by performing brain surgery on the obsessed. It is now fixed via lobotomy and is stored in the brain with TRAUMA_RESILIENCE_MAGIC resistance. No more accidentally losing antag status because someone with a mind restoration virus coughed on you.

This means two things. The first thing is that obsession is now more resistant to conventional brain trauma solutions, and should be lost by complete accident less often. The other thing it means is that the usual way of "handling" an obsessed player is now locked behind certain research. To compensate for this unfortunate outcome, the other departments now have their own ways of handling a lovesick crewmate if Medical cannot.

The Security team like to let their tools do the talking, and the Enhanced Interrogation Chamber gives great advice on how to get over someone. Throw your patient into the chamber and let it reshape their thoughts away from their obsession and towards more important things, like being loyal to Nanotrasen.

The Service department can get in touch with a Chaplain if one is staffed. Chaplains on the default religious sect now have one rite they can perform, and just enough favor to perform it once. They can now expel influence from a rite target, to drive out the voices in the patient's head. Just know that getting the voices out of their head is only the first step...

The Science team, seeking to spit in the face of the Chaplain and his religious miracles, can instead manifest the voices in a more "scientific" manner. If there are voices in the patient's head, simply get rid of the entire head (with a HARS mutator, specifically).

As for Engineering, their method for helping a lovesick crewmate get over someone is the most tried and true. When in doubt, ingest multiple medical substances at once (specifically BZ and Healium, at the same time) and lie on the ground until you can't remember their name.

Cargo I really could not think of anything. I considered a lavaland mushroom soup that only ashwalkers knew how to cook that would calm the mind and cure the trauma, or something, but nothing I tried sat right with me. I came to the conclusion that they're probably just going to get thrown in a lava lake anyways, and that it doesn't really matter. Maybe I'll get back to it in another PR. 

In the event that none of these solutions are employable at the moment, you're on your own. Roleplay it out, or something.

Big thanks to Boondocks for basically telling me how to do the atmospherics part of this PR <3.

## Why It's Good For The Game

First and foremost, this prevents Obsessed from being cured by complete accident and wasting threat. Rejoice.

Due to its rarity and unique objectives, Obsessed leads to some very unique stories. Closing out most of these tales with "but then they got arrested, got brain surgery, and everything went inconsequentially back to normal" doesn't really feel appropriate. Giving some variety, as well as introducing ways for things to go terribly wrong even after the Obsessed has been apprehended, will hopefully make things more interesting. Having different outcomes with differing consequences (for both the victim and crew alike) and difficulty gives a reason to go for different methods depending on the situation. 

## Changelog

:cl:
add: Adds several new methods of removing someone's obsession brain trauma!
add: Adds the Obsession Ghost. A hateful, powerful, and slippery ghost that acts as a physical manifestation of an Obsessed crewmate's uncontrolled thoughts. It will flee the area when attacked unless anchored to the area by lit candles.
add: Adds the Exorcism rite, which can manifest an obsessed crewmember's thoughts into the physical realm as an Obsession Ghost. Defeat it to break its influence over their mind!
add: HARS can also be used to do that, but in a way less cool and more destructive manner.
add: Tying someone down and forcing them to huff a bunch of BZ and Healium will make them completely forget about their obsession, removing the trauma.
qol: Obsessed antagonist status is no longer liable to being revoked immediately after acquisition (usually the result of mind restoration viruses).
balance: Obsessed brain trauma can no longer be removed with standard brain surgery. It now requires a lobotomy, and removing the trauma will cause additional guaranteed damage to the patient's brain.
balance: Obsessed brain trauma now is resistant to all but a few methods of removal.
/:cl:

### CODE REVIEW

**Comment by itseasytosee** [comment]
Really cool. The exorcism ritual could have applications elsewhere.

**Comment by Mothblocks** [comment]
@tralezab

**Comment by DemoCrack** [comment]
CHRIST COMPELS YE.

File: code/datums/brain_damage/creepy_trauma.dm [hunk 1]
```diff
@@ -69,6 +69,8 @@
 	owner.mind.remove_antag_datum(/datum/antagonist/obsessed)
 	if(obsession)
 		UnregisterSignal(obsession, COMSIG_MOB_EYECONTACT)
+	owner.AdjustUnconscious(30)
```
Comment (Fikou) [inline code/datums/brain_damage/creepy_trauma.dm:72]: seconds defines

File: code/game/machinery/hypnochair.dm [hunk 2]
```diff
@@ -133,6 +133,9 @@
 	REMOVE_TRAIT(victim, TRAIT_DEAF, HYPNOCHAIR_TRAIT)
 	if(!(victim.get_eye_protection() > 0))
 		victim.cure_trauma_type(/datum/brain_trauma/severe/hypnotic_trigger, TRAUMA_RESILIENCE_SURGERY)
+		if(victim.has_trauma_type(/datum/brain_trauma/special/obsessed))
+			victim.cure_trauma_type(/datum/brain_trauma/special/obsessed, TRAUMA_RESILIENCE_MAGIC)
+			to_chat(victim, span_warning("Your mind feels empty as the [name] stops. It feels like you've forgotten about something very important."))
```
Comment (Fikou) [inline code/game/machinery/hypnochair.dm:138]: just do [src], not "the [name]"

File: code/modules/mob/living/simple_animal/hostile/retaliate/ghost.dm [hunk 3]
```diff
@@ -64,3 +64,47 @@
...
+		var/list/around = view(src, vision_range)
+		for(var/obj/item/candle/anchor in around)
+			if(anchor.lit)
+				if(prob(20)) //spamming the message would get annoying
+					src.visible_message(span_warning("The [src] attempts to flee through the astral plane, but is confined to this realm by the [anchor]!"), span_warning("You channel your might to escape into the astral plane, but are confined by the [anchor]!"))
+				return
+
+		var/turf/destination = find_safe_turf(extended_safety_checks = TRUE)
+		for(var/obj/machinery/light/flick in get_area(src))
+			flick.flicker(10)
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/hostile/retaliate/ghost.dm:105]: is this time? if yes use seconds defines

File: code/modules/religion/rites.dm [hunk 4]
```diff
@@ -747,3 +747,68 @@
 	..()
 	to_chat(user, span_nicegreen("You feel [GLOB.deity]'s tenacity pouring into you!"))
 	user.AddElement(/datum/element/tenacious)
+
+
+///exorcism rite
+
+/datum/religion_rites/exorcism //In the future I would love to turn exorcism (and also posession) into a component or something more versatile, but for now this works.
+	name = "Exorcism"
+	desc = "The first thing they teach you in Chaplain College. Expels hostile spirits, demonic influence, and any other junk that may be floating around in someone's soul."
```
Comment (Fikou) [inline code/modules/religion/rites.dm:756]: this is really misleading. anyone's first guess is gonna be this curing cultists

**Comment by blackdav123** [comment]
exorcism would be cool as an alternate way to cure cultists

**Comment by VladinXXV** [comment]
No one likes getting a permanent brain trauma. The best method here seems to be HARS, which is a little strange to me. Professional surgery should be more effective than losing your head (literally).  I also think there's room to allow the Psychologist to help here, maybe tied to their skillchip in some way?

**Comment by itseasytosee** [comment]
> exorcism would be cool as an alternate way to cure cultists

Might be fun to have a heretic interaction too. Not De-antaging or anything, but you could do some fun stuff.

**Comment by ATH1909** [comment]
> It is now fixed via lobotomy and is stored in the brain with TRAUMA_RESILIENCE_MAGIC resistance.

This really doesn't sit right with me- we literally have a TRAUMA_RESILIENCE_LOBOTOMY resilience tier for stuff that's supposed to require something lobotomy-grade or stronger to cure.

**Comment by Rhials** [comment]
> > It is now fixed via lobotomy and is stored in the brain with TRAUMA_RESILIENCE_MAGIC resistance.
> 
> This really doesn't sit right with me- we literally have a TRAUMA_RESILIENCE_LOBOTOMY resilience tier for stuff that's supposed to require something lobotomy-grade or stronger to cure, but you're using TRAUMA_RESILIENCE_MAGIC here?

Yeah, I figured someone would bring this up eventually.

In its current state, the obsession trauma has TRAUMA_RESILIENCE_SURGERY. Despite this, it is cured by many things that were probably not intended, such as brain healing nanites (rest in peace) and mind restoration viruses. That's fine with other brain traumas, but when it comes to one that dictates your antag status (and consumes dynamic threat) it should not be as vulnerable to circumvention. Despite it being removed by a lobotomy, extra care needs to be taken to ensure that it isn't accidentally removed by anything that can also heal lobotomies (like changeling panacea)

If you don't agree with that -- Just think of it as an antag status, represented and handled through a magic brain trauma that can only be removed by a few unique methods. One of those methods just so happens to be a lobotomy.

**Comment by ATH1909** [comment]
> Despite TRAUMA_RESILIENCE_LOBOTOMY being the intended resilience for removal via lobotomy, extra care needs to be taken to ensure that it isn't accidentally removed by anything that isn't a lobotomy (like changeling panacea, or anything else that may be implemented in the future)

TBH, being obsessed does sound like something the panacea power should be able to cure. It's a 20 chemical "make whatever this bullshit is stop" button (that you have to actively press), no?

> If you don't agree with that -- Just think of it as an antag status, represented and handled through a magic brain trauma that can only be removed by a few unique methods. One of those methods just so happens to be a lobotomy.

yeah, but this kind of thing is what generates those weird-ass myths/common misinformation pitfalls that just REFUSE to die. do you have any idea how many people still believe that knockdown punches require targeting the mouth??

**Comment by ATH1909** [comment]
Update: I did a quick search, and it turns out that the only things in the game (that I could find) that cure specifically lobotomy-grade traumas or lower are lobotomies (duh) and the resistance 9 threshold effect of the mind restoration symptom. That's it (unless you count, like, direct proc calls from admins or something). Everything else either doesn't cure lobotomy-grade traumas or can also cure magic-grade traumas.

**Comment by tralezab** [comment]
Sorry, I've been having deep thoughts about this. In a way, it's somewhat not the vision I had for obsessed, but I think it's an interesting way to take it. In terms of storymaking, there's a lot of juice thrown in here, and at the end of the day, one of the biggest goals for obsessed for me was making that storytelling happen. That's why I added picture objectives, and spending time around the target, etc etc. So tenatively I say yeah, let's see what happens.

**Comment by OneAsianTortoise** [comment]
> Update: I did a quick search, and it turns out that the only things in the game (that I could find) that cure specifically lobotomy-grade traumas or lower are lobotomies (duh) and the resistance 9 threshold effect of the mind restoration symptom. That's it. Unless you count, like, direct proc calls from admins or something, everything else either doesn't cure lobotomy-grade traumas or can also cure magic-grade traumas.

I remember brain washing (literally, washing brain with wash machine and skillchip) can cure up to lobotomy tier trauma, but I am probably wrong though.

**Comment by Mothblocks** [comment]
We have a lot of open PRs and it doesn't look like any maintainer has expressed interest in this so far. If you still want this ask around for a maintainer willing to push it

**Comment by Rhials** [comment]
@tralezab since you sounded pretty hesitant on allowing it already, I figure you’re the only maintainer I should be approaching for a merge on this

**Pull request closed without merging**